code/modules/atmospherics/machinery/components/unary_devices/thermomachine.dm

```diff
@@ -109,25 +109,32 @@
 
 /obj/machinery/atmospherics/components/unary/thermomachine/process_atmos()
 	..()
-	if(!on || !nodes[1])
+	if(is_operational)
```

Comment:
The var itself should be added to the !on || !nodes[1] bit

Reply:
i dont know, i didnt want to mess with the code itself because i know literally nothing about atmos, i was just trying to fix the bug without trying to mess with any code related to atmos. i can try it tomorrow to see if that works

Reply:
atmos isn't all that spooky, if you need help feel free to ask tho, I get why it might be worrying